### PR TITLE
Update ConfigMap docs

### LINKED ISSUE #4509: Update ConfigMap Docs
**This is a...** 

- [x ] Feature Request
- [ ] Bug Report

Take a look these two topics to see whether they are redundant. If so, combine into one topic. If you decide that we need two topics, prune the content so that information is not repeated.
* [Configure Containers Using a ConfigMap](https://kubernetes.io/docs/tasks/configure-pod-container/configmap/)
* [Use ConfigMap Data in Pods](https://kubernetes.io/docs/tasks/configure-pod-container/configure-pod-configmap/)

### PR DESCRIPTION
**Pull request by spzala**
Currently we have two diferent doc pages to create ConfigMaps and configuring pod using ConfigMaps. There are redundancy between these two docs and also it creates some
confusion using the doc. Also the current separated pages structure doesn't fit with how rest of the pages structured under this section. Merging these two docs into single page and removing
redundancy etc.

fixes #4509 

> ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> For 1.9 Features: set Milestone to `1.9` and Base Branch to `release-1.9`
> ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> NOTE: Please check the “Allow edits from maintainers” box (see image below) to
> [allow reviewers to fix problems](https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/) on your patch and speed up the review process.
>
> Please delete this note before submitting the pull request.
>
> NOTE: After opening the PR, please *un-check and re-check* the "Allow edits from maintainers" box. This is a temporary workaround to address a known issue with GitHub.

Allow edits from maintainers checkbox

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/website/6648)

### CODE REVIEW

**Comment by spzala** [comment]
@chenopis @steveperry-53  I could better organized the flow while removing some 40+ redundant lines by merging the content of two files. If you think the page is long then I can take out ```How to create ConfigMaps..``` kind of doc and put it under ```/docs/concepts/configuration``` under a new entry called ```ConfigMaps``` otherwise as a user it looks good to me to have how to create and use ConfigMaps in a same page.  Thanks!

**Comment by spzala** [comment]
@zacharysarah @chenopis @steveperry-53  a gentle reminder, if this can be reviewed now. Thanks!

**Comment by steveperry-53** [comment]
I'm starting a review now.

**Comment by tengqm** [comment]
The merge looks good to me. I'm wondering if we need an entry in the concepts section. ConfigMap might be not so familiar to new users.

**Comment by steveperry-53** [comment]
@spzala Thanks for doing this; I think it looks good. I put one comment in the text.

@spzala @tengqm @chenopis 
One question: I don't understand this statement. Do you?
"Kubelet doesn't support the use of ConfigMaps for pods not found on the API server. This includes every pod created using kubectl or indirectly via a replication controller."

It seems to be saying that if you want to use a ConfigMap, you can't use kubectl to create your Pods.

**Comment by steveperry-53** [comment]
@tengqm I agree that eventually we will want a concept topic for ConfigMap.

File: docs/tasks/configure-pod-container/configure-pod-configmap.md [hunk 1]
```diff
@@ -284,6 +482,32 @@ When a ConfigMap already being consumed in a volume is updated, projected keys a
 
 ## Understanding ConfigMaps and Pods
 
+ConfigMaps allow you to decouple configuration artifacts from image content to keep containerized applications portable.
```
Comment (steveperry-53) [inline docs/tasks/configure-pod-container/configure-pod-configmap.md]: I think we could remove this **Understanding** section. It repeats ideas and exact text from earlier in the topic.

**Comment by tengqm** [comment]
@steveperry-53 Good question.

>Kubelet doesn't support the use of ConfigMaps for pods not found on the API server. 
    This includes every pod created using kubectl or indirectly via a replication controller. 
    It does not include pods created via the Kubelet's `--manifest-url` flag, `--config` flag, or the Kubelet REST API. 

May be this should be rewritten to something like:

Kubelet doesn't support the use of ConfigMaps for Pods not found on the API server.
This includes pods created via the Kubelet's `--manifest-url` flag, `--config` flag, or the Kubelet REST API.
In other words, ConfigMaps can be used only in Pods directly created using `kubectl` or indirectly created by a controller such as a Deployment.

**Comment by spzala** [comment]
@steveperry-53 good question and agree that it's confusing. ConfigMaps sure can be used with pods created via kubectl - direct or via deployment. So I like the way @tengqm has described. Making changes addressing this and your other comment on removing a redundant sub-section of `Understaing ConfigMaps` Thank you both!
